micrometer-core/src/test/java/io/micrometer/core/instrument/MultiGaugeTest.java

```diff
@@ -162,6 +162,52 @@ public Meter.Id map(Meter.Id id) {
         testOverwrite("prefix.my.multi.gauge");
     }
 
+    @Test
+    void withMeterFilterIgnoreTags() {
+        registry.config().meterFilter(MeterFilter.ignoreTags("ignored"));
+
+        MultiGauge multiGauge = MultiGauge.builder("mg").register(registry);
+
+        multiGauge.register(List.of(Row.of(Tags.of("key", "1", "ignored", "1"), 1d)));
+        assertThat(registry.get("mg").tag("key", "1").gauges()).hasSize(1);
+        assertThat(registry.get("mg").tag("key", "1").gauge().value()).isEqualTo(1d);
+
+        multiGauge.register(List.of(Row.of(Tags.of("key", "1", "ignored", "2"), 2d)));
+        assertThat(registry.get("mg").tag("key", "1").gauges()).hasSize(1);
+        assertThat(registry.get("mg").tag("key", "1").gauge().value()).isEqualTo(1d);
+
+        multiGauge.register(List.of(Row.of(Tags.of("key", "1", "ignored", "3"), 3d)), true);
+        assertThat(registry.get("mg").tag("key", "1").gauges()).hasSize(1);
+        assertThat(registry.get("mg").tag("key", "1").gauge().value()).isEqualTo(3d);
+    }
+
+    @Test
+    void withMeterFilterMapTags() {
+        registry.config()
+            .meterFilter(
+                    MeterFilter.replaceTagValues("key", value -> Integer.parseInt(value) % 2 == 0 ? "even" : "odd"));
```

Comment:
Is this second test added testing a different code path? As far as I can tell it is testing the same code path as the above test. I don't think we need both. They both test the case where ID `a` which maps to `c` is registered followed by ID `b` which also maps to `c` being subsequently registered.

Reply:
The code path in `MultiGauge` and `MeterRegistry` is the same, the use-case/transformation is different: the previous test removes a tag, this one modifies its value so the result is the same (same set of tags), how we get there is different.
I think it's ok to remove it.